EventBrite/Drivers/EventBriteFieldDisplayDriver.cs

```diff
@@ -0,0 +1,188 @@
+
+        #region Dependencies
+
+        private IEventBriteSettingsService _eventBriteSettingsService;
+        private readonly IHttpClientFactory _clientFactory;
+
+        #endregion Dependencies
+
+        #region Constructor
+
+        public EventBriteFieldDisplayDriver(IStringLocalizer<EventBriteFieldDisplayDriver> localizer, IEventBriteSettingsService eventBriteSettingsService, IHttpClientFactory clientFactory)
+        {
+            T = localizer;
+            _eventBriteSettingsService = eventBriteSettingsService;
+            _clientFactory = clientFactory;
+        }
+
+        #endregion Constructor
+
+        #region ContentFieldDisplayDriver<EventBriteField>
+
+        #region Display
+
+        public override IDisplayResult Display(EventBriteField field, BuildFieldDisplayContext context)
+        {
+            return Initialize<DisplayEventBriteFieldViewModel>(GetDisplayShapeType(context), model =>
+            {
+                model.Field = field;
+                model.Part = context.ContentPart;
+                model.PartFieldDefinition = context.PartFieldDefinition;
+            })
+            .Location("Content")
+            .Location("SummaryAdmin", "");
+        }
+
+        #endregion Display
+
+        #region Edit
+
+        public override IDisplayResult Edit(EventBriteField field, BuildFieldEditorContext context)
+        {
+            return Initialize<EditEventBriteFieldViewModel>(GetEditorShapeType(context), model =>
+            {
+                model.Field = field;
+                model.Part = context.ContentPart;
+                model.PartFieldDefinition = context.PartFieldDefinition;
+
+                model.Data = JsonConvert.SerializeObject(field.EventBriteData == null ? new EventBriteItem() : field.EventBriteData);
+            });
+        }
+
+        public override async Task<IDisplayResult> UpdateAsync(EventBriteField field, IUpdateModel updater, UpdateFieldEditorContext context)
+        {
+            var model = new EditEventBriteFieldViewModel();
+
+            await updater.TryUpdateModelAsync(model, Prefix, m => m.Data);
+
+            try
+            {
+                var settings = await _eventBriteSettingsService.GetSettingsAsync();
+
+                if (string.IsNullOrWhiteSpace(settings.PrivateToken))
+                {
+                    updater.ModelState.AddModelError("Value", "Eventbrite API settings are not set correctly, please check these and try again.");
+                }
+
+                var url = GetUrl(updater, model);
+
+                var eventBriteEventDto = await GetEventBriteEventDtoAsync(updater, settings, url);
+
+                EventBriteVenueDto eventBriteVenueDto = await GetEventBriteVenueDtoAsync(updater, settings, eventBriteEventDto);
+
+                if (updater.ModelState.ErrorCount > 0)
+                {
+                    return Edit(field, context);
+                }
+
+                if (eventBriteVenueDto != null)
```

Comment:
I'd change this if block to a predicate.

```
field.EventBriteData = eventBriteVenueDto != null ? new EventBriteItem(eventBriteEventDto, eventBriteVenueDto) : new EventBriteItem(eventBriteEventDto);
```

Looking at what the code is doing, I'm wondering if the null check is even needed in this section. Why not move that to the constructor for `EventBriteItem`? Moves unnecessary logic out the driver.

```
field.EventBriteData = new EventBriteItem(eventBriteEventDto, eventBriteVenueDto);
```

If `eventBriteVenueDto` is `null` then the constructor can handle it.

Reply:
You're right, the only reason I had different constructors was in case the venue dto was null, but I've moved that logic into the constructor instead, as the driver shouldn't have to care.